File: CS/DXSample/ViewModels/BarViewModel.cs
using System.Collections.ObjectModel;

namespace DXSample.ViewModels {
    public class BarViewModel {
        public BarViewModel() {
            Items = new ObservableCollection<BarItemViewModelBase>();
        }
        public BarType BarType { get; set; }
        public string Caption { get; set; }
        public ObservableCollection<BarItemViewModelBase> Items { get; set; }
    }
    public enum BarType { MainMenu, StatusBar };
}

File: VB/DXSample/ViewModels/BarViewModel.vb
Imports System.Collections.ObjectModel

Namespace DXSample.ViewModels

    Public Class BarViewModel

        Public Sub New()
            Items = New ObservableCollection(Of BarItemViewModelBase)()
        End Sub

        Public Property BarType As BarType

        Public Property Caption As String

        Public Property Items As ObservableCollection(Of BarItemViewModelBase)
    End Class

    Public Enum BarType
        MainMenu
        StatusBar
    End Enum
End Namespace
